Assets/UI/launchbar_CQUI.lua

```diff
@@ -0,0 +1,223 @@
+include( "GameCapabilities" );
```

Comment:
Do we need this? The base game LauchBar.lua includes it and the files that include this file should all include the base game file.

Reply:
I was trying to figure out the right method for doing this.  Basically, it's an order of operations thing - and I guess there's 2 ways to go about it?  I've seen it be done both ways, I don't know which is better, or if it's necessary to include other things.  Here's my thought process on it:

**BaseGame file and Extension File**
-BaseFileFoo_CQUI_basegame.lua
```
include("BaseFileFoo")  -- the basegame file
include("BaseFileFoo_CQUI.lua") -- our extension/replacer file
```
BaseFileFoo_CQUI.lua
```
include("WhateverThingsTheBaseFileHad?")
```
Whatever globals are defined in _BaseFileFoo_ are available in BaseFileFoo_CQUI because the CQUI file loads after.  The locals in _BaseFileFoo_ are not available outside of that file, though.  

**Just the CQUI Extension File**
BaseFileFoo_CQUI.lua
```
include ("BaseFileFoo")
```

I haven't been able to figure out why do the first one instead of the second, but there was at least one scenario where trying only the second style didn't seem to work and I had to do the one with the BaseFileFoo_CQUI_basegame...

---

Not sure if the extension file (in this case launchbar_CQUI) refers to anything in GameCapabilities, so perhaps in this case it's not needed...

Reply:
I think the approach for the first way is better. It allows us to have one file `BaseFileFoo_CQUI.lua` that we can use for all version (Base/Exp1/Exp2). This file contains the actual mod changes. So if we need to extend/replace something that the expansion also extends/replaces, then I think we need the first option.

With the second option it becomes more complex when you need to include `BaseFileFoo_Expansion1`, since that file will most likely already include `BaseFileFoo`. Double includes might result in some lines of code being run more than once (unverified guess), like those that exist outside a function. So i think the second option is only viable if the expansions don't do a ReplaceUIScript with the context in question.

Regarding your comment about
`include("WhateverThingsTheBaseFileHad?")`
I don't think it would be needed since the second way doesn't need it I don't see why the first way would. Both include `include("BaseFileFoo")  -- the basegame file` and should have access to the same things. We can also just remove `include("WhateverThingsTheBaseFileHad?")` and see if we get Lua errors for trying to use undefined functions. I'm concerned about double includes (where the `BaseFileFoo` includes the things we try to include in `BaseFileFoo_CQUI`), in my mind the kind of errors that might come up from doing this would be ones that are subtle and hard to debug.